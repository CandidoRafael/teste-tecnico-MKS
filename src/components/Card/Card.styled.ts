import { styled  } from 'styled-components'

export const CardContainer = styled.section`
    box-shadow: rgba(50, 50, 105, 0.149) 0px 2px 5px 0px,
    rgba(0, 0, 0, 0.05) 0px 1px 1px 0px;
    border-radius: 8px;
    background-color: #fff;
    display: flex;
    flex-direction: column;
    align-items: center;
    
`

export const CardImage = styled.img`
    width: 13.2rem;
    height: 220px;
    object-fit: contain;
`

export const CardTitle = styled.article`
    display: flex;
    align-items: flex-start;
    gap: 5px;

    h1 {
        font-size: 16px;
        width: 124px;
        height: 38px;
        font-style: normal;
        font-weight: 400;
    }

    span {
        background-color: #373737;
        color: #fff;
        padding: .3rem;
        border-radius: 5px;
        font-size: 15px;
        font-weight: 700;
    }
`

export const CardDesc = styled.p`
    margin: .4rem 0 .6rem 0;
    font-weight: 300;
    font-size: 10px;
    padding: .5rem;
`

export const CardButton = styled.button`
    display: flex;
    align-items: center;
    cursor: pointer;
    justify-content: center;
    gap: 5px;
    width: 100%;
    border: none;
    background-color: var(--color-blue);
    height: 32px;
    color: #fff;
    font-size: 14px;
    font-style: normal;
    font-weight: 600;
    text-transform: uppercase;
    border-radius: 0 0 8px 8px;
    transition: .3s all linear;

    &:hover {
        background-color: #1e66d3;
    }

    &:disabled {
        cursor: not-allowed;
    }
`