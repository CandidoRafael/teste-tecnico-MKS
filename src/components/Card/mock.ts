export const cardPropsMock = {
    id: 1,
    name: 'Produto 1',
    brand: 'Marca Produto',
    description: 'Descrição Produto',
    price: 3000,
    photo: 'img/img.png',
    quantity: 2
}