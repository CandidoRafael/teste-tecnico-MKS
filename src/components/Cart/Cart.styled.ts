import styled from 'styled-components'

export const CartWrapper = styled.div`
    background-color: #fff;
    border-radius: 5px;
    position: relative;
    display: flex;
    justify-content: space-around;
    align-items: center;
    color: #000000;
    margin-bottom: .5rem;
  
    img {
        width: 5rem;
        padding: .2rem;
        border-radius: 5px;

        @media (max-width: 520px) {
            width: 3rem;
        }
    }

    h2 {
     width: 113px;
     display: flex;
     justify-content: center;
     font-size: 13px;
    font-weight: 400;

    @media (max-width: 520px) {
        font-size: 10px;
    }

    }
`

export const CartQtd = styled.div`
    display: flex;
    flex-direction: column;

    span {
        font-size: 7px;
        font-weight: 400;
    }

    div{
        display: flex;
        gap: 4px;
        border-radius: 4px;
        border: 0.3px solid #BFBFBF;
        background: #FFF;

        p {
            font-size: 10px;
            display: flex;
            align-items: center;
        }

        button:nth-child(1) {
            border: none;
            border-right: 1px solid #BFBFBF;
            cursor: pointer;
            padding: 4px;
            background-color: transparent;
        }

        button:nth-child(3) {
            border: none;
            border-left: 1px solid #BFBFBF;
            cursor: pointer;
            padding: 4px;
            background-color: transparent;
        }
    }
`

export const CartPrice = styled.span`
    font-size: 14px;
    font-weight: 700;
    color: #000000;
    border-radius: 5px;
    padding: .2rem;

    @media (max-width: 520px) {
        font-size: 12px;
    }
`

export const CartClose = styled.i`
    position: absolute;
    right: -5px;
    top: -5px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 18px;
    height: 18px; 
    background-color: #000000;
    border-radius: 50%;
    color: #fff;
    cursor: pointer;

    svg {
        display: flex;
        width: 16px;
    }
`