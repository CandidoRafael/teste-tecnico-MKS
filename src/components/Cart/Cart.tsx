import useStore from "../../store/store";
import { Product } from "../../types/Product"
import { CartClose, CartPrice, CartQtd, CartWrapper } from "./Cart.styled"
import CloseIcon from '@mui/icons-material/Close';

const Cart = ({ product }: { product: Product }) => {

    const { 
      removeItemFromCart,
      decreaseQuantity,
      increaseQuantity
    } = useStore()

    return (
    <CartWrapper>

     <CartClose onClick={() => removeItemFromCart(product.id)}>
      <CloseIcon />
     </CartClose>

     <img src={product.photo}  alt="imagem produto" />
     <h2>{product.name}</h2>

     <CartQtd>
      <span>Qtd:</span>
       <div>
        <button onClick={() => decreaseQuantity(product)}>-</button>
         <p>{product.quantity}</p>
         <button onClick={() => increaseQuantity(product)}>+</button>
       </div>
     </CartQtd>

     <CartPrice>
        R${product.price}
     </CartPrice>

    </CartWrapper>
  )
}

export default Cart