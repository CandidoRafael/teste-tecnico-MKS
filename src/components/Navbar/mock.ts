export const useStoreMock = jest.fn().mockReturnValue({
  toggleSidebar: jest.fn(),
  cart: [],
});