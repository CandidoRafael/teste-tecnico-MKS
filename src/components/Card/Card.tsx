import { Product } from "../../types/Product"
import { TextLimit } from "../TextLimit/TextLimit"
import { CardButton, CardContainer, CardImage, CardTitle } from "./Card.styled"

import useStore from "../../store/store";
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';
import AddShoppingCartIcon from '@mui/icons-material/AddShoppingCart';

const Card = ({ product }: { product: Product }) => {

  const { setCart, cart } = useStore()
  
  return (
    <CardContainer key={product.id}>
      <CardImage src={product.photo} alt="imagem produto" />
      <CardTitle>
       <h1>{product.name}</h1>
       <span>R${product.price}</span>
      </CardTitle>
    
      <TextLimit text={product.description} limit={89} />

    
      <CardButton onClick={() => setCart(product)} disabled={cart.some((item) => item.id === product.id)}>
        {cart.some((item) => item.id === product.id) ? (
          <>
            <AddShoppingCartIcon />
            Adicionado
          </>
        ) : (
          <>
          <ShoppingBagIcon />
          Comprar
          </>
        )}
      </CardButton>

    </CardContainer>
  )
}

export default Card