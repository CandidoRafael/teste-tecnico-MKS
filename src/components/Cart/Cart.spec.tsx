import { render, screen  } from '@testing-library/react';
import { cardPropsMock } from '../Card/mock';
import Cart from './Cart'; 

const mockProduct = cardPropsMock

describe('<Cart />', () => {
  it('should render the Cart correctly', () => {
    render(<Cart product={mockProduct} />);
    
    expect(screen.getByText('Produto 1')).toBeInTheDocument();
    expect(screen.getByAltText('imagem produto')).toBeInTheDocument()
    expect(screen.getByText('Qtd:')).toBeInTheDocument()
    expect(screen.getByText('R$3000')).toBeInTheDocument()

    expect(screen.getByRole('button', {name: '-'})).toBeInTheDocument()
    expect(screen.getByRole('button', {name: '+'})).toBeInTheDocument()
  });

  it('should match snapshot', () => {
    const { container } = render(<Cart product={mockProduct} />)
    expect(container.firstChild).toMatchSnapshot()
 })

});
