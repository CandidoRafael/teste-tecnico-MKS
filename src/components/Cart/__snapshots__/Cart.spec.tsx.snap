// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cart /> should match snapshot 1`] = `
<div
  class="sc-bdfCDU kDxWUw"
>
  <i
    class="sc-hKgKIp ehZBwG"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </i>
  <img
    alt="imagem produto"
    src="img/img.png"
  />
  <h2>
    Produto 1
  </h2>
  <div
    class="sc-gsTDqH fPmuRT"
  >
    <span>
      Qtd:
    </span>
    <div>
      <button>
        -
      </button>
      <p>
        2
      </p>
      <button>
        +
      </button>
    </div>
  </div>
  <span
    class="sc-dlfmHC eojmPb"
  >
    R$
    3000
  </span>
</div>
`;
